### ISSUE TITLE
Duplicated values when appending to array

### ISSUE THREAD

**Issue by crisbal** [mikefarah/yq#1145]
**Describe the bug**

Input:

```yaml
- children: ["c1"]
- children: []
- children: []
- children: ["c1", "c2"]
```
Query:

```
.[] | .children += ["a"]
```

Result:

```yaml
children: ["c1", a, a, a, a]
children:
  - a
  - a
  - a
  - a
children:
  - a
  - a
  - a
  - a
children: ["c1", "c2", a, a, a, a]
```

Expected result:

```yaml
- children: ["c1", "a"]
- children: ["a"]
- children: ["a"]
- children: ["c1", "c2", "a"]
```

Version of yq: 4.22.1
Operating system: linux
Installed via: binary

---

jq behaviour

``` json
yq -o json /tmp/bug.yml | jq '.[] | .children += ["a"]'
{
  "children": [
    "c1",
    "a"
  ]
}
{
  "children": [
    "a"
  ]
}
{
  "children": [
    "a"
  ]
}
{
  "children": [
    "c1",
    "c2",
    "a"
  ]
}
```

**Comment by crisbal** [comment]
The following works instead.

```
>>> yq '.[].children += ["a"]' /tmp/bug.yml
- children: ["c1", a]
- children:
    - a
- children:
    - a
- children: ["c1", "c2", a]
```

**Comment by crisbal** [comment]
More interesting use case

Input:

```yaml
- type: "skip"
  children: ["c1"]
- type: "skip"
  children: []
- type: "ok"
  children: []
- type: "ok"
  children: ["c1", "c2"]
```
Query:

```
yq 'map(select(.type != "skip").children += ["a"])' /tmp/bug.yml
```

Output:

```yaml
- type: "skip"
  children: ["c1"]
- type: "skip"
  children: []
- type: "ok"
  children:
    - a
    - a
    - a
    - a
- type: "ok"
  children: ["c1", "c2", a, a, a, a]
```

**Comment by mikefarah** [comment]
Thanks for raising with a clear example, definitely a bug, looking in to it.

**Comment by mikefarah** [comment]
Fixed in 4.23.1

**Comment by crisbal** [comment]
Thanks for the quick fix! Really appreciate it. I will give it a try on monday.
